p2p/src/net/types/services.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use serialization::{Decode, Encode};
+
+#[derive(Eq, PartialEq, Clone, Copy, Debug)]
```

Comment:
I think you should add here `#[repr(u64)]` so that the underlying integers are u64, not u32.

Reply:
I checked a large flag value and it works normally without it. Added it anyway.